Meter.c

```diff
@@ -277,50 +428,166 @@ static void GraphMeterMode_draw(Meter* this, int x, int y, int w) {
       GraphMeterMode_pixPerRow = PIXPERROW_ASCII;
    }
 
-   attrset(CRT_colors[METER_TEXT]);
-   int captionLen = 3;
-   mvaddnstr(y, x, this->caption, captionLen);
-   x += captionLen;
-   w -= captionLen;
-
    struct timeval now;
    gettimeofday(&now, NULL);
    if (!timercmp(&now, &(data->time), <)) {
       struct timeval delay = { .tv_sec = CRT_delay/10, .tv_usec = (CRT_delay-((CRT_delay/10)*10)) * 100000 };
       timeradd(&now, &delay, &(data->time));
 
-      for (int i = 0; i < nValues - 1; i++)
-         data->values[i] = data->values[i+1];
+      data->drawOffset = (this->pl->settings->alignGraphMeter) ?
+                         (data->drawOffset + 1) % 2 : 0;
 
-      char buffer[nValues];
-      Meter_updateValues(this, buffer, nValues - 1);
+      for (int i = 0; i < GRAPH_NUM_RECORDS - 1; i++) {
+         data->values[i] = data->values[i + 1];
+         memcpy(&(data->colors[i * data->colorRowSize]),
+                &(data->colors[(i + 1) * data->colorRowSize]),
+                sizeof(*data->colors) * data->colorRowSize);
+      }
+
+      char buffer[GRAPH_NUM_RECORDS];
+      Meter_updateValues(this, buffer, GRAPH_NUM_RECORDS - 1);
 
-      double value = 0.0;
       int items = this->curItems;
-      for (int i = 0; i < items; i++)
-         value += this->values[i];
-      value /= this->total;
-      data->values[nValues - 1] = value;
+
+      double *stack1 = data->stack1;
+      double *stack2 = data->stack2;
+      stack1[0] = stack2[0] = 0.0;
+      for (int i = 0; i < items; i++) {
+         stack1[i + 1] = stack2[i + 1];
+         stack2[i + 1] = stack2[i] + this->values[i];
+      }
+      // May not assume this->full be constant. (Example: Swap meter when user
+      // does swapon/swapoff.)
+      stack1[items + 1] = stack2[items + 1];
+      stack2[items + 1] = this->full;
+      data->values[GRAPH_NUM_RECORDS - 1] = stack2[items];
+      double scale;
+      if (isPercentGraph) {
+         data->values[GRAPH_NUM_RECORDS - 1] /= this->full;
+         if (stack1[items + 1] != this->full && stack1[items + 1] > 0) {
+            for (int i = 0; i < items; i++) {
+               stack1[i + 1] = stack1[i + 1] * this->full / stack1[items + 1];
+            }
+         }
```

Comment:
As we're not talking hundrets of stacked values, this should be negligible …

```suggestion
         if (stack1[items + 1] > 0) {
            scale = this->full / stack1[items + 1];
            for (int i = 0; i < items; i++) {
               stack1[i + 1] = stack1[i + 1] * scale;
            }
         }
```